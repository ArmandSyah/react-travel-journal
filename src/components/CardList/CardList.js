import React from "react";

import data from "../../data/data";
import Card from "./Card/Card";

export default function CardList(){
    const generateCard = (cardData, index) => {
        const card = <Card key={cardData.title} {...cardData} />  

        if (data.length - 1 == index){
            return card;
        }

        return (
            <>
                {card}
                <hr />
            </>
        )
    }
    
    return (
        <div className="cardlist">
            {data.map(generateCard)}
        </div>
    )
}