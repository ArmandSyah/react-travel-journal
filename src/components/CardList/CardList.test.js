import React from "react";
import { render, cleanup } from "@testing-library/react";

import CardList from './CardList';

afterEach(cleanup);

it('renders all the cards in the list', async () => {
    const { getAllByTestId } = render(<CardList />)

    const cards = getAllByTestId("card");
    expect(cards).toHaveLength(3);
});

it('renders', async () => {
    const { asFragment } = render(<CardList />)
    expect(asFragment()).toMatchSnapshot();
});