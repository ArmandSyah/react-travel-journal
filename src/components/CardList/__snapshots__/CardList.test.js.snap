// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <div
    class="cardlist"
  >
    <div
      class="card"
      data-testid="card"
    >
      <img
        class="card--image"
        src="https://source.unsplash.com/WLxQvbMyfas"
      />
      <div
        class="cardinfo"
      >
        <div
          class="cardinfo--location"
        >
          <i
            class="fa-solid fa-location-dot"
          />
          <span>
            JAPAN
          </span>
          <a
            href="https://goo.gl/maps/1DGM5WrWnATgkSNB8"
          >
            View On Google Maps
          </a>
        </div>
        <h1>
          Mount Fuji
        </h1>
        <div
          class="cardinfo--maintext"
        >
          <span
            class="cardinfo--maintext--date"
          >
            12 Jan, 2021 - 24 Jan, 2021
          </span>
          <span>
            Mount Fuji is the tallest mountain in Japan, standing at 3,776 meters (12,380 feet). Mount Fuji is the single most popular tourist site in Japan, for both Japanese and foreign tourists.
          </span>
        </div>
      </div>
    </div>
    <hr />
    <div
      class="card"
      data-testid="card"
    >
      <img
        class="card--image"
        src="https://source.unsplash.com/JmuyB_LibRo"
      />
      <div
        class="cardinfo"
      >
        <div
          class="cardinfo--location"
        >
          <i
            class="fa-solid fa-location-dot"
          />
          <span>
            AUSTRALIA
          </span>
          <a
            href="https://goo.gl/maps/1DGM5WrWnATgkSNB8"
          >
            View On Google Maps
          </a>
        </div>
        <h1>
          Sydney Opera House
        </h1>
        <div
          class="cardinfo--maintext"
        >
          <span
            class="cardinfo--maintext--date"
          >
            27 May, 2021 - 8 Jun, 2021
          </span>
          <span>
            The Sydney Opera House is a multi-venue performing arts centre in Sydney. Located on the banks of the Sydney Harbour, it is often regarded as one of the 20th century's most famous and distinctive buildings
          </span>
        </div>
      </div>
    </div>
    <hr />
    <div
      class="card"
      data-testid="card"
    >
      <img
        class="card--image"
        src="https://source.unsplash.com/3PeSjpLVtLg"
      />
      <div
        class="cardinfo"
      >
        <div
          class="cardinfo--location"
        >
          <i
            class="fa-solid fa-location-dot"
          />
          <span>
            NORWAY
          </span>
          <a
            href="https://goo.gl/maps/1DGM5WrWnATgkSNB8"
          >
            View On Google Maps
          </a>
        </div>
        <h1>
          Geirangerfjord
        </h1>
        <div
          class="cardinfo--maintext"
        >
          <span
            class="cardinfo--maintext--date"
          >
            01 Oct, 2021 - 18 Nov, 2021
          </span>
          <span>
            The Geiranger Fjord is a fjord in the Sunnmøre region of Møre og Romsdal county, Norway. It is located entirely in the Stranda Municipality.
          </span>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
